crates/sui-core/src/unit_tests/move_integration_tests.rs

```diff
@@ -1832,6 +1833,23 @@ fn test_entry_point_string_vec_error() {
     })
 }
 
+#[test]
+#[cfg_attr(msim, ignore)]
+fn test_object_struct_error() {
+    run_tokio_test_with_big_stack(async move {
+        let mut build_config = BuildConfig::default();
+        build_config.config.test_mode = true;
+        let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        // in this package object struct is defined incorrectly and publishing should fail (it's
+        // defined in test-only code hence cannot be checked by transactional testing framework
+        // which goes through "normal" publishing path which excludes tests).
+        path.push("src/unit_tests/data/object_struct/");
+        let res = sui_framework::build_move_package(&path, build_config);
+        eprintln!("ERR: {}", res.as_ref().err().unwrap());
+        assert!(matches!(res.err().unwrap(), SuiError::ExecutionError(_)));
```

Comment:
Can we case on which one?

Reply:
To an extent - we can see if the `ExecutionError` string parameter contains a certain keyword. Will do!